### PR TITLE
feat(polars): implement `.sql` methods

### PR DESCRIPTION
**Pull request by cpcloud**
This PR implements `con.sql` and `Table.sql` support for the polars backends. ~A lot of functionality~ subqueries are not implemented yet (and we generate a lot of them), but the various `test_dot_sql.py` tests should start to `XPASS` once support is implemented upstream.

### CODE REVIEW

**Comment by kszucs** [comment]
@cpcloud needs a rebase

File: ibis/backends/polars/__init__.py [hunk 1]
```diff
@@ -224,8 +239,10 @@ def create_table(
         overwrite: bool = False,
     ) -> ir.Table:
         if schema is not None and obj is None:
-            raise NotImplementedError(
-                "Empty table creation is not yet supported in the Polars backend"
+            import pyarrow as pa
+
+            obj = pl.LazyFrame(
+                pa.Table.from_pylist([], schema=schema_to_pyarrow(schema))
```
Comment (kszucs) [inline ibis/backends/polars/__init__.py:245]: You can pass a schema to `LazyFrame` directly, then `pl.LazyFrame([], schema=polars_schema)` should be sufficient without importing pyarrow.

File: ibis/backends/polars/compiler.py [hunk 2]
```diff
@@ -1006,70 +1006,70 @@ def bitwise_not(op):
...
     if func is None:
         raise com.OperationNotDefinedError(f'{type(op).__name__} not supported')
     return func(left, right)
 
 
 @translate.register(ops.ElementWiseVectorizedUDF)
-def elementwise_udf(op):
-    func_args = list(map(translate, op.func_args))
+def elementwise_udf(op, **kw):
+    func_args = list(map(partial(translate, **kw), op.func_args))
```
Comment (kszucs) [inline ibis/backends/polars/compiler.py:1020]: Can you expand it into a list comprehension?

**Review by kszucs — APPROVED** [review]
Just nits other than that it looks good to me.

File: ibis/backends/polars/compiler.py (hunk 2 above)
Comment (cpcloud) [reply ibis/backends/polars/compiler.py:1020]: Sure!

File: ibis/backends/polars/__init__.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/polars/__init__.py:245]: Nice 🎉

File: ibis/backends/polars/compiler.py (hunk 2 above)
Comment (cpcloud) [reply ibis/backends/polars/compiler.py]: Done.

**Pull request merged**